src/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -151,12 +200,39 @@ protected void WriteBytes(byte b1, byte b2)
             _offset += 2;
         }
 
+        protected Task WriteBytesAsync(byte b1, byte b2)
+        {
+            if (_offset + 1 >= bufferLength)
+            {
+                return FlushAndWriteBytesAsync(b1, b2);
+            }
+            else
+            {
+                _buffer[_offset++] = b1;
+                _buffer[_offset++] = b2;
+                return Task.CompletedTask;
+            }
+        }
+
+        private async Task FlushAndWriteBytesAsync(byte b1, byte b2)
+        {
+            await FlushBufferAsync().ConfigureAwait(false);
+            _buffer[_offset++] = b1;
+            _buffer[_offset++] = b2;
+        }
```

Comment:
Yeah, just wanted to keep the async implementations consistent with the sync version.